doc/2/guides/main-concepts/3-querying/index.md

```diff
@@ -8,3 +8,604 @@ order: 300
+
+:::
+
+::: Csharp
+
+```csharp
+SearchOptions options = new SearchOptions();
+options.Scroll = "5s";
+
+SearchResults result = await kuzzle.Document.SearchAsync(
+    "ktm-open-data",
+    "thamel-taxi",
+    JObject.Parse(@"{
+      ""query"": {
+        ""term"": {
+          ""city"": ""Antalya""
+        }
+      }
+    }"),
+    options);
+
+while (result) {
+  Console.Out.WriteLine(result.hits);
+  result = result.NextAsync();
+}
+```
+
+:::
+
+::::
+
+### Paginate with `from` and `size`
+
+Pagination can be done by incrementing the `from` parameter value to retrieve further results.
+
+It's the fastest pagination method available, but also the less consistent.
+
+::: info
+Because this method does not freeze the search results between two calls, there can be missing or duplicated documents between two result pages.  
+Also it's not possible to retrieve more than 10000 documents with this method.
+:::
+
+**Example:** _Use sort and size to navigate through pages of results_
+
+```bash
+kourou document:search ktm-open-data thamel-taxi --from 0 --size 2
+
+kourou document:search ktm-open-data thamel-taxi --from 2 --size 2
+```
+
+### Paginate with `search_after`
+
+Pagination can be done by using the [search_after](https://www.elastic.co/guide/en/elasticsearch/reference/7.4/search-request-body.html#request-body-search-search-after) parameter.
+
+This method allows to navigate through result pages by providing values that identify the next documents.
+
+::: warning
+You have to provide a sort combination that will **always identify one item only**. The recommended way is to use the field `_id` which is certain to contain one unique value for each document.
+:::
+
+**Example:** _Use sort and size to navigate through pages of results_
+
+```bash
+kourou sdk:query document:search \
+  -i ktm-open-data -c thamel-taxi -a size=2 --body '{
+    sort: [
+      { _id: "desc" }
+    ],
+  }'
+
+# {
+#   "hits": [
+#     {
+#       "_id": "liaa",
+#         # ...
+#       }
+#     },
+#     {
+#       "_id": "jenow",
+#         # ...
+#       }
+#     }
+#   ],
+#   "total": 4
+# }
+```
+
+Then we will include the `_id` of the last document in the `search_after` parameter:
+```bash
+kourou sdk:query document:search \
+  -i ktm-open-data -c thamel-taxi -a size=2 --body '{
+    sort: [
+      { _id: "desc" }
+    ],
+    search_after: ["jenow"]
+  }'
+
+# {
+#   "hits": [
+#     {
+#       "_id": "domisol",
+#         # ...
+#       }
+#     },
+#     {
+#       "_id": "aschen",
+#         # ...
+#       }
+#     }
+#   ],
+#   "total": 4
+# }
+```
+
+::: info
+Because this method does not freeze the search results between two calls, **there can be missing or duplicated documents between two result pages**.  
+This method efficiently mitigates the costs of scroll searches, but returns less consistent results: it's a middle ground, **ideal for real-time search requests**.  
+Also it's not possible to retrieve more than 10000 documents with this method.
```

Comment:
False: this method is capable of returning as many documents as needed (given that each page is less than 10k documents).

```suggestion
```